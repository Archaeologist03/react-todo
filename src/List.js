import React from 'react';


function List(props) {
    let listMapped = props.listArr.map(item => {
       
        return (
             item ? <li 
            className="listItem"
            key={Math.random()}>
                <span className="listItemName">{item}</span>
                <a href="#" ><span className="listItemDelete" onClick={() => props.deleteItem(item)}>Delete</span></a>
            </li> : null
        ) 
    });
    

    

    return (
        <ul>
            {listMapped}
        </ul>
    )
}


export default List;