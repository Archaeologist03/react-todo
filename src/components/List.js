import React from 'react';

function List(props) {

  let listMapped = props.listArr.map(item => {
    
    return item ? (
      <li className="listItem" key={Math.random()}>
        <span className="listItemName">{item}</span>
        <a className="listDeleteLink listsBtns" href="#">
          <span
            className="listItemDelete"
            onClick={() => props.onDeleteFromList(item)}>
            Delete
          </span>
        </a>
        <a className="listDoneLink listsBtns" href="#">
          <span className="listItemDone" onClick={() => props.onAddToDone(item)}>
            Done
          </span>
        </a>
      </li>
    ) : null;
  });

  return <ul>{listMapped}</ul>;
}

export default List;
