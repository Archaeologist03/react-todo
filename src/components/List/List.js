import React from 'react';
import { connect } from 'react-redux';

import { deleteFromList } from '../../store/actions/listActions';
import { addToDone } from '../../store/actions/doneActions';

import '../../scss/listItems/listItem.scss';

export function List(props) {
  let listMapped = props.listArr.map(item => {
    return item.name ? (
      <li className='listItem' key={item.id || Math.random()}>
        <span className='listItemName'>{item.name}</span>
        <a
          onClick={e => e.preventDefault()}
          className='listDeleteLink listsBtns'
          href='delete'>
          <span
            className='listItemDelete'
            onClick={() => props.onDeleteFromList(item)}>
            Delete
          </span>
        </a>
        <a
          onClick={e => e.preventDefault()}
          className='listDoneLink listsBtns'
          href='done'>
          <span
            className='listItemDone'
            onClick={() => props.onAddToDone(item.name)}>
            Done
          </span>
        </a>
      </li>
    ) : null;
  });

  return <ul className='todoList-container'>{listMapped}</ul>;
}

const mapStateToProps = state => {
  return {
    listArr: state.app.list,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onDeleteFromList: item => dispatch(deleteFromList(item)),
    onAddToDone: doneItem => dispatch(addToDone(doneItem)),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(List);
