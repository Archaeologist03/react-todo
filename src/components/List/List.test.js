// import configureStore from 'redux-mock-store';
// import thunk from 'redux-thunk';
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { List } from './List';

describe('List', () => {
  it('renders List without crashing given the required props', () => {
    const props = {
      listArr: [],
      onDeleteFromList: jest.fn(),
      onAddToDone: jest.fn(),
    };
    const wrapper = shallow(<List {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render listItemName element with task name as children', () => {
    const props = {
      listArr: [{ name: 'task1', id: 1 }],
      onDeleteFromList: jest.fn(),
      onAddToDone: jest.fn(),
    };
    const wrapper = shallow(<List {...props} />);
    const listItemNameClassElement = wrapper.find('.listItemName');
    expect(listItemNameClassElement.props().children).toBe(
      props.listArr[0].name,
    );
  });

  it('should call onDeleteFromList fn from props on click', () => {
    const props = {
      listArr: [{ name: 'task1', id: 1 }],
      onDeleteFromList: jest.fn(),
      onAddToDone: jest.fn(),
    };
    const wrapper = shallow(<List {...props} />);
    const listItemDeleteClassElement = wrapper.find('.listItemDelete');

    listItemDeleteClassElement.simulate('click');
    expect(props.onDeleteFromList).toBeCalled();
  });

  it('should call onAddToDone fn from props on click with name arg', () => {
    const props = {
      listArr: [{ name: 'task1', id: 1 }],
      onDeleteFromList: jest.fn(),
      onAddToDone: jest.fn(),
    };
    const wrapper = shallow(<List {...props} />);
    const listItemDoneClassElement = wrapper.find('.listItemDone');

    listItemDoneClassElement.simulate('click');
    expect(props.onAddToDone).toHaveBeenCalledWith(props.listArr[0].name);
  });
});
