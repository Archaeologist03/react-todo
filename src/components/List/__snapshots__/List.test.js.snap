// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders List without crashing given the required props 1`] = `
<ul
  className="todoList-container"
/>
`;
